internal/datastreams/processor.go

```diff
@@ -172,12 +172,17 @@ type kafkaOffset struct {
 	timestamp  int64
 }
 
+type bucketKey struct {
+	serviceName string
+	btime       int64
+}
+
 type Processor struct {
 	in                   *fastQueue
 	hashCache            *hashCache
 	inKafka              chan kafkaOffset
-	tsTypeCurrentBuckets map[int64]bucket
-	tsTypeOriginBuckets  map[int64]bucket
+	tsTypeCurrentBuckets map[bucketKey]bucket
+	tsTypeOriginBuckets  map[bucketKey]bucket
```

Comment:
@piochelepiotr Is it possible that changing the key of the map may lead to some performance degradation?

Beyond this, the PR looks fine to me. Remember to create this same PR against `v2-dev`. Thanks!

Reply:
I think it should be fine, since these maps are very small (max two timestamps given that we flush every 10 seconds).

I would be more worried if I had updated the map inside the `bucket` struct.